xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -67,37 +79,36 @@ public XdsServerBuilder xdsServingStatusListener(
     return this;
   }
 
-  /**
-   * Unsupported call. Users should only use {@link #forPort(int, ServerCredentials)}.
-   */
-  @DoNotCall("Unsupported. Use forPort(int, ServerCredentials) instead")
-  public static ServerBuilder<?> forPort(int port) {
-    throw new UnsupportedOperationException(
-        "Unsupported call - use forPort(int, ServerCredentials)");
+  public static XdsServerBuilder forPort(int port) {
```

Comment:
Hmmm... allowing this with `InsecureServerCredentials` as the default? I remember discussions where we wanted users to explicitly pass plaintext creds and not defaulting to plaintext. That's also why we have `Grpc.newServerBuilderForPort(int, ServerCredentials)` but not `Grpc.newServerBuilderForPort(int)`. Also the super class `ForwardingServerBuilder.forPort(int)` is still marked as `DoNotCall` so this looks suspicious

Reply:
I see, now I know why it was previously like this `"Unsupported call - use forPort(int, ServerCredentials)"` we should keep it like that then.

Reply:
> Also the super class ForwardingServerBuilder.forPort(int) is still marked as DoNotCall so this looks suspicious

Subclasses can still re-define it without problems. We have the DoNotCall places where the subclass _doesn't_ redefine it, just because it is super confusing if a user does end up calling it. But that shouldn't be used as an argument here.

But yes, for new clients/servers we probably want to encourage the Credentials approach as it is more clear. I don't see a need to have the non-credential method here.